submissions/TaniaBondarenko/friends-app/app.js

```diff
@@ -0,0 +1,177 @@
+
+function makeContainerEmpty() {
+  let innerText = " ";
+  FRIENDS.innerHTML = innerText.replace(innerText, " ");
+}
+
+function showSorteredFriends({ target }) {
+  switch (target.value) {
+    case "nameUp":
+      showSortByNameUp();
+      break;
+    case "nameDown":
+      showSortByNameDown();
+      break;
+    case "ageUp":
+      showSortByAgeUp();
+      break;
+    case "ageDown":
+      showSortByAgeDown();
+      break;
+  }
+}
+
+function showSortByNameUp() {
+  makeContainerEmpty();
+  if (MALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameUp().filter((el) => el.gender === "male");
+  } else if (FEMALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameUp().filter((el) => el.gender === "female");
+  } else if (ALL_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameUp();
+  }
+  addFriends(sortedFriends);
+}
+
+function showSortByNameDown() {
+  makeContainerEmpty();
+  if (MALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameDown().filter((el) => el.gender === "male");
+  } else if (FEMALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameDown().filter((el) => el.gender === "female");
+  } else if (ALL_RADIO_BUTTON.checked) {
+    sortedFriends = sortByNameDown();
+  }
+  addFriends(sortedFriends);
+}
+
+function showSortByAgeUp() {
+  makeContainerEmpty();
+  if (MALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeUp().filter((el) => el.gender === "male");
+  } else if (FEMALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeUp().filter((el) => el.gender === "female");
+  } else if (ALL_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeUp();
+  }
+  addFriends(sortedFriends);
+}
+
+function showSortByAgeDown() {
+  makeContainerEmpty();
+  if (MALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeDown().filter((el) => el.gender === "male");
+  } else if (FEMALE_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeDown().filter((el) => el.gender === "female");
+  } else if (ALL_RADIO_BUTTON.checked) {
+    sortedFriends = sortByAgeDown();
+  }
+  addFriends(sortedFriends);
+}
+
+function sortByNameUp() {
+  return allFriends.sort((a, b) => a.name.first.localeCompare(b.name.first));
+}
+
+function sortByNameDown() {
+  return allFriends.sort((a, b) => b.name.first.localeCompare(a.name.first));
+}
+
+function sortByAgeUp() {
+  return allFriends.sort((a, b) => a.dob.age - b.dob.age);
+}
+
+function sortByAgeDown() {
+  return allFriends.sort((a, b) => b.dob.age - a.dob.age);
+}
+
+document.querySelector(".filter").addEventListener("click", doFilter);
+
+function doFilter({ target }) {
+  if (target.value === "male" || target.value === "female") {
+    makeContainerEmpty();
+    sortedFriends = filterBySex(target.value);
+    isFiltered = true;
+    addFriends(sortedFriends);
+  } else if (target.value === "all") {
+    makeContainerEmpty();
+    addFriends(allFriends);
+    isFiltered = false;
+  }
```

Comment:
1. `isFiltered === !isFiltered`
2. `makeContainerEmpty` repeated twice
3. `if (target.value === "all") ... else` - just reduce redunandant code

Reply:
I've rewrote function, but when I've tried to replace `isFiltered=true` with `!isFiltered` (the default state of `isFiltered=false`), it broke all my sorting logic, so I leaved it as it was.